binderhub/repoproviders.py

```diff
@@ -91,6 +100,17 @@ def is_banned(self):
                 return True
         return False
 
+    def is_whitelisted(self):
```

Comment:
I think this is the method that we should test first. I'd instantiate the repo provider class, set the `spec` attribute by hand (if it isn't easy to do the normal way because repo provider makes network calls) and then test if `is_whitelisted()` gives the right answer for different spec/whitelist combos.

Reply:
We can reuse a lot of the code from https://github.com/jupyterhub/binderhub/blob/b5ccb6c6b5259d53bc0f2a062cfbe290afe71544/binderhub/tests/test_repoproviders.py#L66-L92 for this I think